File: app/src/main/java/com/tgk/app/core/PinyinTranslator.kt
package com.tgk.app.core

object PinyinTranslator {
    fun translate(code: String, lang: String): String {
        var result = code
        GrammarManager.grammarMap.forEach { (pinyin, target) ->
            val replacement = if (target.contains(" ")) target.split(" ")[0] else target
            result = result.replace(pinyin, replacement)
        }
        return when (lang) {
            "python" -> result.replace("zhen", "True").replace("jia", "False")
            "javascript" -> result.replace("dayin", "console.log")
            else -> result
        }
    }
}

File: MauiUI/Core/PinyinTranslator.cs
using System;
using System.Collections.Generic;

namespace TianGongKaiWu.Core;
public class PinyinTranslator : IDisposable
{
    private readonly Dictionary<string, string> _grammar;
    public PinyinTranslator()
    {
        _grammar = GrammarManager.GrammarDict;
    }
    public string Translate(string pinyinCode, string targetLang)
    {
        if (string.IsNullOrEmpty(pinyinCode)) return string.Empty;
        string result = pinyinCode;
        foreach (var pair in _grammar)
        {
            string replacement = pair.Value;
            if (replacement.Contains(' '))
                replacement = replacement.Split(' ')[0];
            result = result.Replace(pair.Key, replacement);
        }
        // 按语言微调
        if (targetLang == "python")
        {
            result = result.Replace("zhen", "True").Replace("jia", "False").Replace("kong", "None");
        }
        else if (targetLang == "javascript")
        {
            result = result.Replace("dayin", "console.log");
        }
        return result;
    }
    public void Dispose() { }
}
